modules/web-discovery-project/sources/url-analyzer.es

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { extractHostname } from "../core/tlds";
+import { parse } from "../core/url";
+import logger from "./logger";
+
+const URL_PATTERNS = [
+  {
+    type: "search-goi",
+    regexp:
+      /^https:[/][/][^/]*[.]google[.].*?[#?&;]((q=[^&]+&([^&]+&)*tbm=isch)|(tbm=isch&([^&]+&)*q=[^&]+))/,
+    prefix: "search?tbm=isch&gbv=1&q=",
+  },
+  {
+    type: "search-gov",
+    regexp:
+      /^https:[/][/][^/]*[.]google[.].*?[#?&;]((q=[^&]+&([^&]+&)*tbm=vid)|(tbm=vid&([^&]+&)*q=[^&]+))/,
+    prefix: "search?tbm=vid&gbv=1&q=",
+  },
+  {
+    type: "search-go",
+    regexp: /^https:[/][/][^/]*[.]google[.].*?[#?&;]/,
+    prefix: "search?q=",
+  },
+  {
+    type: "search-ya",
+    regexp: /^https:[/][/][^/]*[.]search[.]yahoo[.].*?[#?&;][pq]=[^$&]+/,
+    prefix: "search?q=",
+    queryFinder(parsedUrl) {
```

Comment:
nit: consider just passing searchParams instead of parsedUrl

Reply:
I think it's better to pass `parsedUrl` in this case even though in the existing list of URL patterns we extract query from `searchParams` only since the logic here is that you take a URL and extract a search query from it which can potentially be located anywhere in the URL, not only in `searchParams`. This approach looks easier to comprehend to me